src/cpp/security/accesscontrol/DistinguishedName.h

```diff
@@ -0,0 +1,201 @@
+// Copyright 2023 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/*!
+ * @file DistinguishedName.h
+ */
+
+#ifndef _SECURITY_ACCESSCONTROL_DISTINGUISHEDNAME_H_
+#define _SECURITY_ACCESSCONTROL_DISTINGUISHEDNAME_H_
+
+#include <array>
+#include <string>
+
+namespace eprosima {
+namespace fastrtps {
+namespace rtps {
+namespace security {
+
+/*
+ * RFC 1779: https://datatracker.ietf.org/doc/html/rfc1779 specifies how to represent DistinguishedNames
+ * in string format (obsolete in favor of rfc2253).
+ *
+ * RFC 2253: https://datatracker.ietf.org/doc/html/rfc2253 specifies how to represent DistinguishedNames
+ * in string UTF-8 format.
+ *
+ * ASN.1 representation of DistinguishedNames
+ * DistinguishedName ::= RDNSequence
+ * RDNSequence ::= SEQUENCE OF RelativeDistinguishedName
+ * RelativeDistinguishedName ::= SET SIZE (1..MAX) OF
+ * AttributeTypeAndValue
+ * AttributeTypeAndValue ::= SEQUENCE {
+ * type  AttributeType,
+ * value AttributeValue }
+ *
+ * Each RDN in a RDNSequence is separated by ','.
+ * Each AttributeTypeAndValue in a RDN is separated by '+'.
+ * Attribute and Value are separated by '='.
+ *
+ */
+
+/**
+ * @brief class that represents a Distinguished Name
+ *
+ * @note so far this implementation works only as a string
+ * @todo make this a real class with real methods that correctly implement RFC 2253:
+ */
+using DistinguishedName = std::string;
+
+/**
+ * @brief This function compares 2 strings representing DistinguishedNames.
+ *
+ * Follow RFC 2253: https://datatracker.ietf.org/doc/html/rfc2253
+ * The idea here is that comparing 2 DistinguishedNames is not as trivial as comparing 2 strings,
+ * nor comparing its values splitting the string.
+ * The comparison must be a complex function that takes into account many specific details.
+ *
+ * So far, Fast DDS gives 2 strings (in some format that we assume is the correct one) and this compares
+ * whether the two strings refer to the same DistinguishedName.
+ * It would be better (and easier) to store DistinguishedName in a specialized function instead of in a string,
+ * and thus this comparison would be much easier and efficient.
+ * But this will be left for future improvements.
+ *
+ * @warning this function does not correctly fulfilled all requirements of RFC 2253.
+ * Requirements that are not fulfilled:
+ * - multi-attribute
+ * - types accept oid
```

Comment:
Here you may be referring to `void`

```suggestion
 * - types accept void
```